src/java.base/share/classes/com/sun/crypto/provider/CipherBlockChaining.java

```diff
@@ -59,6 +59,11 @@ class CipherBlockChaining extends FeedbackCipher  {
     // variables for save/restore calls
     private byte[] rSave = null;
 
+    // chunkSize is multiple of block size, but otherwise fairly arbitrary
+    // Should be large enough to provide intrinsic with optimization
+    // opportunities
+    private final int chunkSize = 1024*100;
```

Comment:
Shouldn't we call `getBlockSize()` of the parent class here rather than using a hard-coded integer? Variable assignment to be done in constructor in such case.

Reply:
I don't think this constant needs to be dynamic. The reason I mention blocksize, the intrinsic expects multiple of block size (`ArrayUtil.blockSizeCheck(plainLen, blockSize);` assert before-hand), but its otherwise unrelated to the block size. (i.e. "has to be large enough for optimizations").

I believe this class can also be used by DES; iirc, blocksize=8. I suppose I could use `AESConstants.AES_BLOCK_SIZE*6400` as `chunkSize`. If that looks good, I will do that change, though I've seen '16' used as a raw number plenty of times in the package, without 'attribution' to AES_BLOCK_SIZE.

And to be thorough, I should probably put an `ArrayUtil.blockSizeCheck(chunkSize, blockSize);` into the constructor? That way it is checked, but not on every call.

Reply:
Yes, my concern is basically `intrinsic expects multiple of block size`. I would simply declare the variable and than assign it in constructor when `blockSize` is known, no need to call `blockSizeCheck`:

`private final int chunkSize;
.........
 CipherBlockChaining(SymmetricCipher embeddedCipher) {
        super(embeddedCipher);
        k = new byte[blockSize];
        r = new byte[blockSize];
        chunkSize = blockSize * 6400;
    }
`

Reply:
Done